DataFormats/Portable/test/alpaka/test_catch2_portableCollection.dev.cc

```diff
@@ -0,0 +1,121 @@
+#include <catch.hpp>
+
+#include "DataFormats/Portable/interface/PortableCollection.h"
+#include "DataFormats/SoATemplate/interface/SoACommon.h"
+#include "DataFormats/SoATemplate/interface/SoALayout.h"
+#include "DataFormats/SoATemplate/interface/SoAView.h"
+#include "FWCore/Utilities/interface/stringize.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/workdivision.h"
+
+// each test binary is built for a single Alpaka backend
+using namespace ALPAKA_ACCELERATOR_NAMESPACE;
+
+namespace {
+  GENERATE_SOA_LAYOUT(TestLayout, SOA_COLUMN(double, x), SOA_COLUMN(int32_t, id), SOA_SCALAR(uint32_t, num))
+
+  using TestSoA = TestLayout<>;
+
+  constexpr auto s_tag = "[PortableCollection]";
+}  // namespace
+
+TEST_CASE("PortableCollection<T, TDev>", s_tag) {
+  // get the list of devices on the current platform
+  auto const& devices = cms::alpakatools::devices<Platform>();
+  if (devices.empty()) {
+    FAIL("No devices available for the " EDM_STRINGIZE(ALPAKA_ACCELERATOR_NAMESPACE) " backend, "
+         "the test will be skipped.");
+  }
+
+  SECTION("Default constructor") {
+    PortableHostCollection<TestSoA> coll_h;
+    REQUIRE(coll_h.size() == 0);
+    REQUIRE(not coll_h.isValid());
+
+    // Following lines would be undefined behavior, and could lead to crashes
+    //coll->num() = 42;
+    //REQUIRE(coll->num() == 42);
+
+    // CopyToDevice<PortableHostCollection<T>> is not defined
+#ifndef ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED
```

Comment:
These `#ifdef`s could be removed with https://github.com/cms-sw/cmssw/pull/43969